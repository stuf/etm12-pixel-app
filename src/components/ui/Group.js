import * as React from 'karet';

export default function Group({ children, title }) {
  return (
    <section className="group">
      <header className="group__header">{title}</header>

      <div className="group__content">{children}</div>
    </section>
  );
}
