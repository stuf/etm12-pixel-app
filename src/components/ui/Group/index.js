import * as React from 'karet';

import styles from './index.module.scss';

export default function Group({ children, title }) {
  return (
    <section className={styles.root}>
      <header className={styles.header}>{title}</header>

      <div className={styles.content}>{children}</div>
    </section>
  );
}
