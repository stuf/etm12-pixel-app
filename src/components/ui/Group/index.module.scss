@import '~styles/variables';

$group-header-font-weight: 600;

.root {
  border-bottom: dotted 1px $color-gray-400;
}

.header {
  padding: $spacing-base;
  font-weight: $group-header-font-weight;
}

.content {
  padding: $spacing-base;

  > * {
    margin-top: 0;
    margin-bottom: 0;
  }

  > * + * {
    margin-top: $spacing-base;
  }
}
