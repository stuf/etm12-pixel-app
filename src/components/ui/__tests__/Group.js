import * as React from 'karet';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Group from 'components/ui/Group';

it('matches snapshot', () => {
  const wrap = mount(<Group title="Group title">Group content</Group>);
  expect(toJson(wrap)).toMatchSnapshot();
});
