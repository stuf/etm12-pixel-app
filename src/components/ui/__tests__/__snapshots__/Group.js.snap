// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Group
  title="Group title"
>
  <section
    className="group"
  >
    <header
      className="group__header"
    >
      Group title
    </header>
    <div
      className="group__content"
    >
      Group content
    </div>
  </section>
</Group>
`;
